auctionapp/src/main/java/com/internship/auctionapp/controller/BidController.java

```diff
@@ -0,0 +1,26 @@
+package com.internship.auctionapp.controller;
+
+import com.internship.auctionapp.dto.BidDto;
+import com.internship.auctionapp.service.BidService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/api/bids")
+public class BidController {
+    private final BidService bidService;
+
+    public BidController(BidService bidService) {
+        this.bidService = bidService;
+    }
+
+    @PostMapping("/new-bid")
+    @PreAuthorize("hasAnyAuthority('USER', 'ADMIN')")
+    public ResponseEntity<BidDto> register(@RequestBody BidDto bidDto) {
```

Comment:
There is no need for `/new-bid` or method named `register`. `POST /api/bids` is self explanatory. But I'm not sure we should return CREATED status if we do the update of existing bid.